### PR TITLE
Add pulp file ACS refresh command

### PR DESCRIPTION
**Pull request by daviddavis**
[noissue]

### CODE REVIEW

File: pulpcore/cli/common/context.py [hunk 1]
```diff
@@ -148,41 +148,97 @@ def call(self, operation_id: str, non_blocking: bool = False, *args: Any, **kwar
...
-                elif task["state"] == "canceled":
-                    raise click.ClickException("Task canceled")
-                elif task["state"] in ["waiting", "running", "canceling"]:
+                if task_group["all_tasks_dispatched"] is True:
+                    for task in task_group["tasks"]:
+                        task = self.api.call(
+                            "tasks_read", parameters={"task_href": task["pulp_href"]}
+                        )
+                        click.echo(f"Waiting for task {task['pulp_href']}", err=True)
+                        self._poll_task(task, timeout)
```
Comment (mdellweg) [inline pulpcore/cli/common/context.py:226]: This is not accounting the tasks timeout as part of the whole timeout. Maybe counting seconds is not the best approach and we should be starting to compare wall clock times.

File: pulpcore/cli/core/task.py [hunk 2]
```diff
@@ -107,6 +108,6 @@ def cancel(
         try:
             pulp_ctx.wait_for_task(entity)
         except Exception as e:
-            if str(e) != "Task canceled":
+            if not re.match("Task /pulp/api/v3/tasks/[-0-9a-f]*/ canceled", str(e)):
```
Comment (mdellweg) [inline pulpcore/cli/core/task.py:111]: +1

File: pulpcore/cli/file/context.py [hunk 3]
```diff
@@ -23,6 +24,13 @@ class PulpFileACSContext(PulpEntityContext):
     CREATE_ID = "acs_file_file_create"
     UPDATE_ID = "acs_file_file_partial_update"
     DELETE_ID = "acs_file_file_delete"
+    REFRESH_ID = "acs_file_file_refresh"
```
Comment (mdellweg) [inline pulpcore/cli/file/context.py:27]:
```suggestion
    REFRESH_ID: ClassVar[str] = "acs_file_file_refresh"
```

**Review by mdellweg — COMMENTED** [review]
I'd like to have one if the ACS experts to judge the necessary test coverage here.

File: pulpcore/cli/common/context.py (hunk 1 above)
Comment (daviddavis) [reply pulpcore/cli/common/context.py:226]: One question I had is whether timeout should apply to each task or the entire task group?

File: pulpcore/cli/common/context.py [hunk 4]
```diff
@@ -148,41 +148,97 @@ def call(self, operation_id: str, non_blocking: bool = False, *args: Any, **kwar
             click.echo(f"Started background task {task_href}", err=True)
             if not non_blocking:
                 result = self.wait_for_task(result)
+        if isinstance(result, dict) and ["task_group"] == list(result.keys()):
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/common/context.py:151]: No action needed here.
Thinking further: If we could pass this as some kind of annotation to the result without breaking the `call` interface backwards incompatible ...

File: pulpcore/cli/common/context.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/cli/common/context.py:226]: That is indeed a valid question. I do not know.

Comment (mdellweg) [reply pulpcore/cli/common/context.py]: I agree completely. In the best case we can shield the user from thinking about tasks. So it's one operation that takes time.

**Comment by daviddavis** [comment]
@pavelpicka can you review?

**Review by pavelpicka — APPROVED** [review]
LGTM
just "[cli-bad-repo](https://github.com/pulp/pulp-cli/pull/377/files#diff-c3b3f2f409d8e4fcecd980edb6314e598fd501fe9cc3c3bdf2bf6b1ad35e5c05R14)" in tests invokes malformed repo, this case sounds like incomplete or metadata only (if we expect success)

**Comment by daviddavis** [comment]
I renamed the repo `cli-repo-manifest-only` and added a failing refresh test.

**Review by mdellweg — APPROVED** [review]

**Comment by mdellweg** [comment]
Are the necessary bits for this merged to pulpcore and pulp_file or do we need to wait with merging to not break their respective ci?

**Comment by daviddavis** [comment]
> Are the necessary bits for this merged to pulpcore and pulp_file or do we need to wait with merging to not break their respective ci?

The only thing this relies on is the pulp_file refresh command which got merged: https://github.com/pulp/pulp_file/pull/536

**Pull request merged**